usersession/agent/rest_api_test.go

```diff
@@ -903,3 +907,150 @@ func (s *restSuite) TestPostCloseRefreshNotification(c *C) {
+	si := snap.Info{
+		SideInfo: snap.SideInfo{
+			RealName: snapName,
+		},
+		Apps: make(map[string]*snap.AppInfo, 5),
+	}
+	return &si
+}
+
+func addAppToSnap(c *C, snapinfo *snap.Info, app string, isService bool, icon string) {
+	newInfo := snap.AppInfo{
+		Snap: snapinfo,
+		Name: app,
+	}
+	if isService {
+		newInfo.Daemon = "daemon"
+	}
+	snapinfo.Apps[app] = &newInfo
+	createDesktopFile(c, &newInfo, icon)
+}
+
+func (s *restSuite) TestGuessAppIconNoIconPrefixEqualApp(c *C) {
+	si := createSnapInfo("app1")
+	addAppToSnap(c, si, "app1", false, "")
+	icon := agent.GuessAppIcon(si)
+	c.Check(icon, Equals, "")
+}
+
+func (s *restSuite) TestGuessAppIconNoIconPrefixDifferentApp(c *C) {
+	si := createSnapInfo("snap1")
+	addAppToSnap(c, si, "app1", false, "")
+	icon := agent.GuessAppIcon(si)
+	c.Check(icon, Equals, "")
+}
+
+func (s *restSuite) TestGuessAppIconPrefixDifferentApp(c *C) {
+	si := createSnapInfo("snap1")
+	addAppToSnap(c, si, "app1", false, "iconname")
+	icon := agent.GuessAppIcon(si)
+	c.Check(icon, Equals, "iconname")
+}
+
+func (s *restSuite) TestGuessAppIconPrefixEqualApp(c *C) {
+	si := createSnapInfo("app1")
+	addAppToSnap(c, si, "app1", false, "iconname1")
+	addAppToSnap(c, si, "app2", false, "iconname2")
+	icon := agent.GuessAppIcon(si)
+	c.Check(icon, Equals, "iconname1")
+}
+
+func (s *restSuite) TestGuessAppIconServicePrefixEqualApp(c *C) {
+	si := createSnapInfo("app1")
+	addAppToSnap(c, si, "app1", true, "iconname")
+	icon := agent.GuessAppIcon(si)
+	c.Check(icon, Equals, "")
+}
+
+func (s *restSuite) TestGuessAppIconServicePrefixDifferentApp(c *C) {
+	si := createSnapInfo("snap1")
+	addAppToSnap(c, si, "app1", true, "iconname")
+	icon := agent.GuessAppIcon(si)
+	c.Check(icon, Equals, "")
+}
+
+func (s *restSuite) TestGuessAppIconServiceTwoApps(c *C) {
+	si := createSnapInfo("app1")
+	addAppToSnap(c, si, "app1", true, "iconname1")
+	addAppToSnap(c, si, "app2", false, "iconname2")
+	icon := agent.GuessAppIcon(si)
+	c.Check(icon, Equals, "iconname2")
+}
+
+func (s *restSuite) TestGuessAppIconServiceTwoAppsServices(c *C) {
+	si := createSnapInfo("app1")
+	addAppToSnap(c, si, "app1", true, "iconname1")
+	addAppToSnap(c, si, "app2", true, "iconname2")
+	icon := agent.GuessAppIcon(si)
+	c.Check(icon, Equals, "")
+}
+
+func (s *restSuite) TestGuessAppIconServiceTwoAppsOneServicePrefixDifferent(c *C) {
+	si := createSnapInfo("snap1")
+	addAppToSnap(c, si, "app1", true, "iconname1")
+	addAppToSnap(c, si, "app2", false, "iconname2")
+	icon := agent.GuessAppIcon(si)
+	c.Check(icon, Equals, "iconname2")
+}
+
+func (s *restSuite) TestGuessAppIconTwoAppsPrefixDifferent(c *C) {
+	si := createSnapInfo("snap1")
+	addAppToSnap(c, si, "app1", false, "iconname1")
+	addAppToSnap(c, si, "app2", false, "iconname2")
+	icon := agent.GuessAppIcon(si)
+	if (icon != "iconname1") && (icon != "iconname2") {
+		c.Fail()
+	}
+}
+
+func (s *restSuite) TestPostCloseRefreshNotificationWithIconDefault(c *C) {
+	dirs.SetRootDir(c.MkDir())
```

Comment:
this needs to be removed to avoid inconsistency, `dirs.SetRootDir` is already called in `SetUpTest` above.

Reply:
Done.